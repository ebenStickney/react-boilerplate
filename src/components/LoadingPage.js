import React from 'react';


const LoadingPage = () => (
  <div className='loader'>
   <img className='loader__image' src='/images/loading_12.gif' />
   <h2> Loading Your Expenses </h2>
  </div>
);

export default LoadingPage;
