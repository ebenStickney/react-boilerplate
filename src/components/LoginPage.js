import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';


 //creating eventhandler for calling startLogin action
export const LoginPage = ({ startLogin }) => {
  return (
    <div className='box-layout'>
      <div className='box-layout__box'>
        <h1 className='box-layout__title'> App Name </h1>
        <p> Tag Line</p>
        <button onClick={startLogin} className='button'> Login With Google </button>
      </div>

    </div>

  );
};

const mapDispatchToProps = (dispatch) => ({
  startLogin: () =>dispatch(startLogin())
})  //setting up prop to call dispatch action.

export default connect(undefined, mapDispatchToProps)(LoginPage); //creaing HOC with connect
