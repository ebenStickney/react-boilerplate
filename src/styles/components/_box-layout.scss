.box-layout {
  background: url('/images/city-bg_edited.jpg');
  height: 100vh;
  display: flex;
  width: 100vw;
  background-size: cover;
  justify-content: center;
  align-items: center;
}

.box-layout__box {
  background: fade-out(white, .15);
  border-radius: 2px;
  text-align: center;
  width: 25rem;
  padding: $l-size $m-size;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
  
}
