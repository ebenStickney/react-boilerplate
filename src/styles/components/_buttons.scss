.button {
  background-color: $google-blue;
  color: white;
  padding: $s-size;
  display: inline-block;
  border: none;
  font-weight: 300;
  font-size: $font-size-large;
  text-decoration: none;
  line-height: 1;
}

.button-link {
  background-color: Transparent;
}

.button-secondary {
  background-color: #888
}
