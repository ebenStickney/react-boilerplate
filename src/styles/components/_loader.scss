.loader {
  height: 100vh;
  width: 100vw;
  align-items: center;
  justify-content: center;
  display: flex;
  flex-direction: column;
  h2 {
    align-items: center;
    justify-content: center;
    margin: 0;

  }
}

.loader__image {
  height: 15rem;
  width: 20rem;
}
