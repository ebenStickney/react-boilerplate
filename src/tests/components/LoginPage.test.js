import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';


test('should render LoginPage snapshot', () => {
  const wrapper = shallow(<LoginPage/>);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on Click', () => {
  const onClickSpy = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={onClickSpy}/>);
  wrapper.find('button').simulate('click');
  expect(onClickSpy).toHaveBeenCalled();
})
