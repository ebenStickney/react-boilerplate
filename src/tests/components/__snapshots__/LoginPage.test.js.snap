// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage snapshot 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
       App Name 
    </h1>
    <p>
       Tag Line
    </p>
    <button
      className="button"
    >
       Login With Google 
    </button>
  </div>
</div>
`;
